src/Core/Utils/Attribs.hpp

```diff
@@ -305,18 +293,18 @@ class AttribManager {
     }
 
     /// Remove attribute by handle, invalidate the handles to this attribute.
-    /// \note: If a new attribute is added, old invalidated handles may lead to
-    ///        the new attribute.
+    /// \warning If a new attribute is added, old invalidated handles may lead to
+    ///          the new attribute.
     /// \note The complexity for removing an attribute is O(log(n)).
     template <typename T>
     void removeAttrib( AttribHandle<T> h ) {
-        const auto& att = getAttrib( h ); // check the attribute exists
+        const auto& att = getAttrib( h );
         removeAttrib( att.getName() );
     }
 
   private:
     /// Access to the list of attributes.
-    /// \note The complexity for acessing the list of attributes is O(n).
+    /// \note The complexity for accessing the list of attributes is O(n).
     Container attribs() const {
```

Comment:
What is the purpose of this method ?

Reply:
This methods filters out the empty slots (nullptr from removal) from the attribs list, so that one can perform the following for loop without caring about nullptr:
```c++
AttribManager m;
...
for ( auto& attr : m.attribs() )
{
    // no need to check whether attr == nullptr
}
```

Reply:
If we use Handles as iterators, we could have private begin, end methods in the AttribManager and private increment methods in the Handles

Reply:
I don't see the link with what I wrote.

Reply:
I bet it will take more than the 4 lines needed for the functor-based loop version.